### PR TITLE
RELENG-587: Add a passwords.yml block for mozillaonline fenix builds

### PR DESCRIPTION
**Pull request by bhearsum**
I think this is actually really straightforward? I opened https://github.com/mozilla-services/cloudops-infra/pull/3181 and pushed a sops change for the needed cloudops parts, but AFAICT all we need to do is add the new certs to the mapping here.

There's not much testing to do against dev, because we don't have a special dep cert for mozillaonline.

### CODE REVIEW

**Comment by bhearsum** [comment]
(looks like these failures are mypy related; we'll need some requirements updates to fix them)

**Comment by escapewindow** [comment]
At least some of the failures are `KeyError: 'AUTOGRAPH_FENIX_MOZILLA_ONLINE_USERNAME'`

**Comment by escapewindow** [comment]
And yes, the scriptworker_client and iscript failures are mypy related.
I'm not sure if we're getting much of a win from mypy, and I know we're getting a hit in productivity for it. I'm ok with disabling, but I'm also ok with a quick fix.

File: signingscript/docker.d/passwords.yml [hunk 1]
```diff
@@ -81,7 +81,14 @@ in:
           - ["https://autograph-external.prod.autograph.services.mozaws.net",
              {"$eval": "AUTOGRAPH_FENIX_USERNAME"},
              {"$eval": "AUTOGRAPH_FENIX_PASSWORD"},
-             ["autograph_apk", "autograph_apk_fennec_sha1"]
+             ["autograph_apk"]
+            ]
+        project:mobile:fenix:releng:signing:cert:dep-mozillaonline-signing:
+          - ["https://autograph-external.prod.autograph.services.mozaws.net",
```
Comment (escapewindow) [inline signingscript/docker.d/passwords.yml]: Hm. I think we want to keep this `dep-signing`, and change the format name to, say, `autograph_apk_mozillaonline` or similar.

It's up to us if we actually want to create another format for dep, or just change the taskgraph configs to use `dep-signing` + `autograph_apk` for mozillaonline dep signing?

File: signingscript/docker.d/passwords.yml [hunk 2]
```diff
@@ -279,6 +286,12 @@ in:
              {"$eval": "AUTOGRAPH_FENIX_PASSWORD"},
              ["autograph_apk"]
             ]
+        project:mobile:fenix:releng:signing:cert:production-mozillaonline-signing:
+          - ["https://autograph-external.prod.autograph.services.mozaws.net",
+             {"$eval": "AUTOGRAPH_FENIX_MOZILLA_ONLINE_USERNAME"},
+             {"$eval": "AUTOGRAPH_FENIX_MOZILLA_ONLINE_PASSWORD"},
+             ["autograph_apk"]
```
Comment (escapewindow) [inline signingscript/docker.d/passwords.yml]: Same here. We do need a separate set of autograph credentials for mozillaonline, so let's keep the `cert:production-signing`, and add a new `autograph_apk_mozillaonline` or similar format.

**Review by escapewindow — CHANGES_REQUESTED** [review]
Let me know if you have questions?
And agreed, this should be pretty quick once we get the cloudops PR merged.

File: signingscript/docker.d/passwords.yml (hunk 2 above)
Comment (escapewindow) [reply signingscript/docker.d/passwords.yml]: (If we did change the cert, we would need to:
- write a ci-config patch to grant the new cert scopes to fenix, and
- update the [CoT restricted scopes](https://github.com/mozilla-releng/scriptworker/blob/master/src/scriptworker/constants.py#L366) to make sure we only use the new production-mozillaonline-signing cert type against the fenix repo)

File: signingscript/docker.d/passwords.yml (hunk 1 above)
Comment (bhearsum) [reply signingscript/docker.d/passwords.yml]: I don't see any reason to use a new cert for dep signing. Is there a reason we need to change the scope name, if we're using the same cert?

File: signingscript/docker.d/passwords.yml (hunk 2 above)
Comment (bhearsum) [reply signingscript/docker.d/passwords.yml]: It sounds like you're saying the reason we need to use the existing certs is to satisfy CoT? If so, that makes sense. (I couldn't find any reason within `signingscript` to require it.)

File: signingscript/docker.d/passwords.yml (hunk 1 above)
Comment (escapewindow) [reply signingscript/docker.d/passwords.yml]: Nope. Maybe just json-e in the signing kind: for depsigning, use `dep-signing` + `autograph_apk` format; if we're release signing, use `production-signing` + `autograph_apk_mozillaonline` or similar.

File: signingscript/docker.d/passwords.yml (hunk 2 above)
Comment (escapewindow) [reply signingscript/docker.d/passwords.yml]: It's mainly for separating out scopes and yes, restricting certain cert scopes to certain repos in CoT. Formats are, I think, just in the upstreamArtifacts and are no longer scopes at all.

File: signingscript/docker.d/passwords.yml (hunk 1 above)
Comment (bhearsum) [reply signingscript/docker.d/passwords.yml]: sgtm; thanks for the pointers.

**Review by escapewindow — APPROVED** [review]
LGTM, thank you!

I think we can roll these out at any point.

**Pull request merged**